File: examples/fortran/multiplyadd/multiplyadd.f90
program multiplyadd_example
    use multiplyadd_impl
    use iso_fortran_env
    use omp_lib

    implicit none

    integer, parameter :: N = 1000
    integer :: i

    real(kind=real32),dimension(N) :: x
    real(kind=real32),dimension(N) :: y

    x(:) = 2.0
    y(:) = 1.0

    call multiplyadd(20.5, x, y, N)

    do i = 1,N
        if(y(i).ne.42) then
            print"(a,f5.2)", "Expected 42 found ", y(i)
            stop
        endif
    end do

end program multiplyadd_example

File: examples/c++/multiplyadd/multiplyadd.cpp
#include <iostream>
#include <vector>

#include "multiplyadd_impl.hpp"

int main() {

  int nsize = 1000;

  std::vector<int> a(nsize);
  std::vector<int> b(nsize);
  std::vector<int> c(nsize);

  std::fill(std::begin(a), std::end(a), 0);
  std::fill(std::begin(b), std::end(b), 2);
  std::fill(std::begin(c), std::end(c), 3);

  multiplyadd(20, &a, &b, &c, nsize);

  for (auto val : a) {
    if (val != 43) {
      std::cout << "Wrong value! Found " << val << " expected 43" << std::endl;
      return EXIT_FAILURE;
    }
  }

  printf("clean up");

  return EXIT_SUCCESS;
}